plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -195,6 +198,35 @@ func (c *nodePlugin) admitPodStatus(nodeName string, a admission.Attributes) err
 	}
 }
 
+func (c *nodePlugin) admitPodEviction(nodeName string, a admission.Attributes) error {
+	switch a.GetOperation() {
+	case admission.Create:
+		// require eviction to an existing pod object
+		eviction, ok := a.GetObject().(*policy.Eviction)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+		// get the existing pod from the server cache
+		existingPod, err := c.podsGetter.Pods(eviction.Namespace).Get(eviction.Name, v1.GetOptions{ResourceVersion: "0"})
+		if errors.IsNotFound(err) {
+			// wasn't found in the server cache, do a live lookup before forbidding
+			existingPod, err = c.podsGetter.Pods(eviction.Namespace).Get(eviction.Name, v1.GetOptions{})
+		}
+		if existingPod == nil {
```

Comment:
both here and in the delete case, I wonder if we should return a NotFound error instead of a Forbidden error if the `err` here is a NotFound error...